EKF/control.cpp

```diff
@@ -133,8 +133,9 @@ void Ekf::controlFusionModes()
 	}
 
 	// We don't fuse flow data immediately because we have to wait for the mid integration point to fall behind the fusion time horizon.
-	// This means we stop looking for new data until the old data has been fused.
-	if (!_flow_data_ready) {
+	// This means we stop looking for new data until the old data has been fused, unless we are not fusing optical flow,
+	// in this case we need to empty the buffer
+	if (!_flow_data_ready || !_control_status.flags.opt_flow) {
```

Comment:
@kamilritz I don't understand this anymore. When we add the flow sample to the ringbuffer, why don't we make sure that the timestamp of that sample is selected such that it is located in the middle of the integration period. Then you can just pop that sample from the buffer and fuse it right away.

Reply:
For flow sensor without IMU, we need to accumulate the delta_angle by integrating the stand alone imu. When would put them in the buffer with the middle point timestamp, the integration would be harder. We want to move the integration to the flow driver level. After such a change we can push them in the buffer with middle point timestamp.

Reply:
@kamilritz Fair enough. I was hoping to be able to get rid of that logic in control.cpp but for now let's leave it.